src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +43,38 @@
      * Андрей - 600
      * София - 900</p>
      */
+    private static final DateTimeFormatter FORMATTER
+            = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+            throws IllegalDateParametersException {
+
+        LocalDate beginDate = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate endDate = LocalDate.parse(dateTo, FORMATTER);
+
+        StringBuilder result =
```

Comment:
let's move variable declaration after we check dates for valid parameters

Reply:
Done!